coderd/jwtutils/jwe.go

```diff
@@ -0,0 +1,125 @@
+package jwtutils
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"time"
+
+	"github.com/go-jose/go-jose/v4"
+	"github.com/go-jose/go-jose/v4/jwt"
+	"golang.org/x/xerrors"
+)
+
+const (
+	encryptKeyAlgo     = jose.A256GCMKW
+	encryptContentAlgo = jose.A256GCM
+)
+
+type EncryptKeyProvider interface {
+	EncryptingKey(ctx context.Context) (id string, key interface{}, err error)
+}
+
+type DecryptKeyProvider interface {
+	DecryptingKey(ctx context.Context, id string) (key interface{}, err error)
+}
+
+// Encrypt encrypts a token and returns it as a string.
+func Encrypt(ctx context.Context, e EncryptKeyProvider, claims Claims) (string, error) {
+	id, key, err := e.EncryptingKey(ctx)
+	if err != nil {
+		return "", xerrors.Errorf("get signing key: %w", err)
+	}
+
+	encrypter, err := jose.NewEncrypter(
+		encryptContentAlgo,
+		jose.Recipient{
+			Algorithm: encryptKeyAlgo,
+			Key:       key,
+		},
+		&jose.EncrypterOptions{
+			Compression: jose.DEFLATE,
+			ExtraHeaders: map[jose.HeaderKey]interface{}{
+				keyIDHeaderKey: id,
+			},
+		},
+	)
+	if err != nil {
+		return "", xerrors.Errorf("initialize encrypter: %w", err)
+	}
+
+	payload, err := json.Marshal(claims)
+	if err != nil {
+		return "", xerrors.Errorf("marshal payload: %w", err)
+	}
+
+	encrypted, err := encrypter.Encrypt(payload)
+	if err != nil {
+		return "", xerrors.Errorf("encrypt: %w", err)
+	}
+
+	serialized := []byte(encrypted.FullSerialize())
+	return base64.RawURLEncoding.EncodeToString(serialized), nil
```

Comment:
Is there a reason you use `CompactSerialize` for JWS and `FullSerialized` + manual base64 encoding here?

The base64 encoding is not the compact base64 encoding, you can see the `ParseEncryptedCompact` here: https://github.com/go-jose/go-jose/blob/main/jwe.go#L286-L330

So there is no dots I do not think if we do this ourselves.

Reply:
Honestly I just copied the implementation verbatim but I agree I don't see why we shouldn't make the change.